packages/flutter/test/widgets/web_image_test.dart

```diff
@@ -0,0 +1,1308 @@
+    final TestHttpRequest testHttpRequest = TestHttpRequest()
+      ..status = 500
+      ..mockEvent = MockEvent('load', web.Event('test error'))
+      ..response = (Uint8List.fromList(kTransparentImage)).buffer;
+
+    httpRequestFactory = () {
+      return testHttpRequest.getMock() as web_shim.XMLHttpRequest;
+    };
+
+    await tester.pumpWidget(
+      WebImage.network('https://www.example.com/images/frame.png'),
+    );
+    // Pump once to create and start resolving the image.
+    await tester.pumpAndSettle();
+
+    // Since the request for the bytes succeeds, this should put an
+    // Image.network (which resolves to a RawWebImage) in the widget tree.
+    expect(find.byType(RawWebImage), findsOneWidget);
+  });
+
+  testWidgets('Verify WebImage does not use disposed handles', (WidgetTester tester) async {
+    final web_shim.HTMLImageElement image100x100 = createTestWebImage(width: 100, height: 100);
+
+    final _TestWebImageProvider imageProvider1 = _TestWebImageProvider('http://example.com/pic1');
+    final _TestWebImageProvider imageProvider2 = _TestWebImageProvider('http://example.com/pic2');
+
+    final ValueNotifier<_TestWebImageProvider> imageListenable = ValueNotifier<_TestWebImageProvider>(imageProvider1);
+    addTearDown(imageListenable.dispose);
+    final ValueNotifier<int> innerListenable = ValueNotifier<int>(0);
+    addTearDown(innerListenable.dispose);
+
+    bool imageLoaded = false;
+
+    await tester.pumpWidget(ValueListenableBuilder<_TestWebImageProvider>(
+      valueListenable: imageListenable,
+      builder: (BuildContext context, _TestWebImageProvider image, Widget? child) => WebImage(
+        image: image,
+        frameBuilder: (BuildContext context, Widget child, int? frame, bool wasSynchronouslyLoaded) {
+          if (frame == 0) {
+            imageLoaded = true;
+          }
+          return LayoutBuilder(
+            builder: (BuildContext context, BoxConstraints constraints) => ValueListenableBuilder<int>(
+              valueListenable: innerListenable,
+              builder: (BuildContext context, int value, Widget? valueListenableChild) => KeyedSubtree(
+                key: UniqueKey(),
+                child: child,
+              ),
+            ),
+          );
+        },
+      ),
+    ));
+
+    imageLoaded = false;
+    imageProvider1.complete(image10x10);
+    await tester.idle();
+    await tester.pump();
+    expect(imageLoaded, true);
+
+    imageLoaded = false;
+    imageListenable.value = imageProvider2;
+    innerListenable.value += 1;
+    imageProvider2.complete(image100x100);
+    await tester.idle();
+    await tester.pump();
+    expect(imageLoaded, true);
+  });
+
+  testWidgets('Verify WebImage resets its RenderWebImage when changing providers', (WidgetTester tester) async {
+    final GlobalKey key = GlobalKey();
+    final _TestWebImageProvider imageProvider1 = _TestWebImageProvider('example.com/pic.jpg');
+    await tester.pumpWidget(
+      Container(
```

Comment:
Not all of the Containers can be changed into SizedBox. For some of the tests we want to check that the RenderObject associated with a key is the RenderObject for the WebImage. But since SizedBox creates its own RenderObject, the key will associate with the SizedBox RenderObject instead.